python/paddle/fluid/distribute_transpiler.py

```diff
@@ -415,29 +419,35 @@ def __append_optimize_op__(op, block):
             else:
                 self._append_pserver_non_opt_ops(block, op)
 
-        append_block = optimize_block
         # append lr decay ops to the child block if exists
+        lr_decay_block = None
         lr_ops = self._get_lr_ops()
         if len(lr_ops) > 0:
+            lr_decay_block = pserver_program.create_block(GLOBAL_BLOCK_IDX)
             for _, op in enumerate(lr_ops):
-                self._append_pserver_non_opt_ops(append_block, op)
-
-            append_block = pserver_program.create_block(append_block.idx)
+                self._append_pserver_non_opt_ops(lr_decay_block, op)
 
         # append op to the current block
-        per_opt_block = append_block
+        per_opt_block = None
+        pre_block_idx = GLOBAL_BLOCK_IDX
```

Comment:
No globals is needed here, just record the latest block id.

Reply:
done